### PR TITLE
Remove saving stores in the manager struct

### PR DESCRIPTION
**Pull request by ray-milkey**

### CODE REVIEW

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -308,7 +217,61 @@ func (m *Manager) Start() error {
 		return err
 	}
 
-	return nil
+	// Creates gRPC server and registers various services; then serves.
+	authorization := false
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:221]: You can define a function like this https://github.com/onosproject/onos-e2t/blob/master/pkg/manager/manager.go#L148 to start NB server and calls it from manager. We will do the same thing for controllers and SB

File: pkg/manager/manager.go [hunk 2]
```diff
@@ -73,27 +68,10 @@ type Config struct {
...
-	networkSnapshotController *controller.Controller
-	deviceSnapshotController  *controller.Controller
-	ModelRegistry             *modelregistry.ModelRegistry
-	TopoChannel               chan *topodevice.ListResponse
-	OperationalStateChannel   chan events.OperationalStateEvent
-	SouthboundErrorChan       chan events.DeviceResponse
-	Dispatcher                *dispatcher.Dispatcher
-	OperationalStateCache     map[topodevice.ID]devicechange.TypedValueMap
-	OperationalStateCacheLock *sync.RWMutex
+	SouthboundErrorChan     chan events.DeviceResponse
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:71]: How these channels used? If we disable tests, I guess we can define them locally for now as well and inject into other packages correct which need them?

File: pkg/manager/manager.go [hunk 3]
```diff
@@ -111,191 +89,122 @@ func (m *Manager) setTargetGenerator(targetGen func() southbound.TargetIf) {
...
+	if err := m.Start(); err != nil {
+		log.Fatal("Unable to run Manager", err)
+	}
+}
+
+// Start starts the manager
+func (m *Manager) Start() error {
 	opts, err := certs.HandleCertPaths(m.Config.CAPath, m.Config.KeyPath, m.Config.CertPath, true)
 	if err != nil {
 		log.Fatal(err)
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:105]: We should return error and log here. We already call Fatal when we call m.Start()

File: pkg/manager/manager.go [hunk 4]
```diff
@@ -111,191 +89,122 @@ func (m *Manager) setTargetGenerator(targetGen func() southbound.TargetIf) {
...
-	m.OperationalStateCacheLock = &sync.RWMutex{}
+	dsp := dispatcher.NewDispatcher()
+	operationalStateCache := make(map[topodevice.ID]devicechange.TypedValueMap)
+	operationalStateCacheLock := &sync.RWMutex{}
 
 	// Start the NetworkChange controller
-	errNetworkCtrl := m.networkChangeController.Start()
+	errNetworkCtrl := networkChangeController.Start()
 	if errNetworkCtrl != nil {
 		log.Error("Can't start controller ", errNetworkCtrl)
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:176]: Same thing here for any calls inside Start, returns error to the caller

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

**Comment by ray-milkey** [comment]
retest this please

File: pkg/manager/manager.go (hunk 2 above)
Comment (ray-milkey) [reply pkg/manager/manager.go]: Must have been there for the tests. Nothing uses it now.

File: pkg/manager/manager.go [hunk 5]
```diff
@@ -73,27 +68,9 @@ type Config struct {
...
-	networkSnapshotController *controller.Controller
-	deviceSnapshotController  *controller.Controller
-	ModelRegistry             *modelregistry.ModelRegistry
-	TopoChannel               chan *topodevice.ListResponse
-	OperationalStateChannel   chan events.OperationalStateEvent
-	SouthboundErrorChan       chan events.DeviceResponse
-	Dispatcher                *dispatcher.Dispatcher
-	OperationalStateCache     map[topodevice.ID]devicechange.TypedValueMap
-	OperationalStateCacheLock *sync.RWMutex
+	TopoChannel             chan *topodevice.ListResponse
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:71]: What about topoChannel and Operational state channel?

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

**Review by adibrastegarnia — APPROVED** [review]

**Pull request merged**